docs/control-structures.rst

```diff
@@ -145,6 +145,41 @@ throws an exception or goes out of gas.
     external functions happen after any changes to state variables in your contract
     so your contract is not vulnerable to a reentrancy exploit.
 
+.. warning::
+    The type of this.g takes a memory argument, even if the function signature uses calldata.
+    This is because for external function pointers, you are only interested in the calling side and not in the called side.
+    For the caller, it does not make sense to even distinguish
+    function (string calldata) external and function (string memory) external.
```

Comment:
The text here can be rephrased to present a more informed situation to the reader. The lines here are [comments](https://github.com/ethereum/solidity/issues/12778#issuecomment-1169917941) from an ongoing discussion. 
A better approach maybe would be to first explain the error (the `calldata` arguments of external function pointers get converted internally to `memory` arguments) then explain the reasons (interest in calling side not called side...). After that, suggestion of how to amend the situation and examples of the problem can be showed to conclude.